corpus/filter.py

```diff
@@ -148,18 +148,23 @@ def run(self):
 
 
 class FilterCorpusBySegmentsJob(Job):
-    def __init__(self, bliss_corpus, segment_file, compressed=False, invert_match=False):
+    __sis_hash_exclude__ = {"delete_empty_recordings": False}
+
+    def __init__(self, bliss_corpus, segment_file, compressed=False, invert_match=False, delete_empty_recordings=False):
         """
         :param Path bliss_corpus:
         :param list[Path]|Path segment_file: a single segment file or a list of segment files
         :param bool compressed:
         :param bool invert_match:
+        :param bool delete_empty_recordings: if true, empty recordings will be removed
         """
         self.bliss_corpus = bliss_corpus
         self.segment_file_list = [segment_file] if isinstance(segment_file, tk.Path) else segment_file
         self.invert_match = invert_match
+        self.delete_empty_recordings = delete_empty_recordings
 
         self.out_corpus = self.output_path("corpus.xml" + (".gz" if compressed else ""))
+        self.out_removed_recordings = self.output_path("removed_recordings.log")
```

Comment:
This output is always created, but the actual file is only created if `delete_empty_recordings` is set to true.

This may lead to errors if some Job depends on the output, the `FilterCorpusBySegmentsJob` is finished - so sisyphus assumes the file is there - but the actual file on disk is not present.

So I propose either to only create the output if the option is set
```suggestion
        if delete_empty_recordings:
            self.out_removed_recordings = self.output_path("removed_recordings.log")
```

or to create the file in any case. (It will be empty then)